src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -556,29 +604,44 @@ impl Net {
                 self.rx_buffer
                     .iovec
                     .write_all_volatile_at(&self.rx_frame_buf[..vnet_hdr_len() + len], 0)?;
+
+                // SAFETY:
+                // * len is never 0
+                // * Maximum size of the packet is 65562 which fits into u32
+                #[allow(clippy::cast_possible_truncation)]
+                let len = unsafe { NonZeroU32::new_unchecked((vnet_hdr_len() + len) as u32) };
```

Comment:
I used `NonZeroU32` to keep the:
```Rust
            match self.read_from_mmds_or_tap() {
                Ok(None) => {
                     ...
                }
                Ok(Some(bytes)) => {
```
match. Otherwise it would be `Ok(0)` and `Ok(n)`. I don't really mind this option, but I guess `NonZeroU32` is a bit more explicit here.
Also `readv` should never return 0 as we only call it with some buffers passed. If `readv` has nothing to read, it will error with `EAGAIN`.

Reply:
Replaced with `u32`.